### PR TITLE
add End game

### PR DESCRIPTION
**Pull request by PillouMatou**
Users can choose if his game will be limited by the time or the score.
Actually only the score is set, need improvements.

### CODE REVIEW

**Comment by kuhlkrein** [comment]
Renomer l'option "limitateur" par "limitateur de temps/de score activé" ou quelque chose de similaire et explicite

**Comment by kuhlkrein** [comment]
@PillouMatou there are some conflicts

**Comment by kuhlkrein** [comment]
draft review:
Time is not modified in GazePlay (but seems to be saved in configuration)
Maybe we should create a playTimeOutTransition and playScoreReacheTransition in oarder to display something different than the wintransition. (at least for time)
We could add an « auto replay » for exemple with cupsandballs, displaying stats with a « click to stop auto replay » overlay when displaying stats

Maybe we should add an icon to the games concerned by time limitation and score limitation

BibouleJump : only refractor

Blocs :
	Should add lilmiterS
	Do we have to end the last party before ending with timeScore ?

Bubble :
	Score looks good
	Time limitation not working?

CakeFactory :

CreamPie :
	Ok pour Time
	Continue button gives black screen after time limitation end

CupsAndBalls :
	We can add limiterS
	Ok for time
	Maybe time limitation should stop during the shuffle of the gobelet ( with a time of 10s for example, the game can finish while the gobelets are still being shuffled )

Divisor :
	Ok pour Time

EggGame :
	Ok pour Time

Labyrinth : 
	OK pour score
	When using continue, the games end after the first try (I guess the Score limiter is not reset)
	limiteur de temps not working

MagicCards :
	OK pour time
	OK pour score

Potion :
	OK for time limitation the first time but after that it ends after the first new game
	OK pour score

Math101 :
	OK pour score
	Limiteur de temps stops at the first game

Memory :
	« Ok » pour score
	« Ok » pour time
	when clicking continue a bravo appears in background for the first new game and restart the game at the end of the bravo

Moles :
	Ok pour temps
	after first time ending if we press continue the game stops after few seconds
	OK pour score

Ninja :
	Ok pour time
	OK pour score	
Order :
	Bof, des fois ça marche pas, de toute façon le jeu est bugué

Race :	
	Ok pour le temps

RushHour :
	L’affichage des stats est lent, il doit y avoir une petit bouche ou un handler qui freine, mais c’est pareil sans les limiteurs (meme pire limite)

Shooter :
	Ok pour le temps
	OK pour score
	par contre on a une erreur au niveau du suivis du canon, il n’est pas orienté au bon endroit avec le regard

SpotTheDifferences :
	Ok with Time
	small bugs sometimes it doesnt want to select

File: gazeplay-games/src/main/java/net/gazeplay/games/creampie/CreamPie.java [hunk 1]
```diff
@@ -29,13 +29,14 @@ public CreamPie(IGameContext gameContext, Stats stats) {
         final RandomPositionGenerator randomPositionGenerator = gameContext.getRandomPositionGenerator();
 
         hand = new Hand();
-        target = new Target(randomPositionGenerator, hand, stats, gameContext, imageLibrary);
+        target = new Target(randomPositionGenerator, hand, stats, gameContext, imageLibrary, this);
         gameContext.getChildren().add(target);
         gameContext.getChildren().add(hand);
```
Comment (kuhlkrein) [inline gazeplay-games/src/main/java/net/gazeplay/games/creampie/CreamPie.java:34]: move this:
```
gameContext.getChildren().add(target);
gameContext.getChildren().add(hand);
```
at the begining of launcher to fix black screen after continue

File: gazeplay-core/src/main/java/net/gazeplay/ui/scenes/configuration/ConfigurationContext.java [hunk 2]
```diff
@@ -1037,6 +1051,92 @@ HBox buildHeatMapColorHBox(Configuration config, Translator translator) {
...
+
+        HBox hbox = new HBox();
+        hbox.setSpacing(5);
+
+        CheckBox limitTime = buildCheckBox(config.getLimiterTProperty());
+
+        I18NText time = new I18NText(translator, "Time(seconds)");
+        time.setFill(Color.WHITE);
+
+        Spinner<Integer> spinnerT = new Spinner<>(3, 180, 90, 1);
```
Comment (kuhlkrein) [inline gazeplay-core/src/main/java/net/gazeplay/ui/scenes/configuration/ConfigurationContext.java:1064]: Spinner<Integer> spinnerT = new Spinner<>(3, 180, config.getLimiterTime(), 1);

File: gazeplay-core/src/main/java/net/gazeplay/ui/scenes/configuration/ConfigurationContext.java [hunk 3]
```diff
@@ -1037,6 +1051,92 @@ HBox buildHeatMapColorHBox(Configuration config, Translator translator) {
...
+
+        HBox hbox = new HBox();
+        hbox.setSpacing(5);
+
+        CheckBox limitScore = buildCheckBox(config.getLimiterSProperty());
+
+        I18NText score = new I18NText(translator, "score");
+        score.setFill(Color.WHITE);
+
+        Spinner<Integer> spinnerS = new Spinner<>(3, 180, 90, 1);
```
Comment (kuhlkrein) [inline gazeplay-core/src/main/java/net/gazeplay/ui/scenes/configuration/ConfigurationContext.java:1107]: Spinner<Integer> spinnerS = new Spinner<>(3, 180, config.getLimiterScore(), 1);

**Comment by kuhlkrein** [comment]
All games have been reviewed (at least 3 times), now it should be good but if @christophezei or @Adrieennn want to give a new look to it, it would be with pleasure ^^

File: gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv [hunk 4]
```diff
@@ -239,6 +240,10 @@ HeatMapSettings,,,眼动追踪地图设置,,,HeatMap Settings,,Paramètres de la
 AOISettings,,,关键注视区设置,,,A.O.I Settings,,Paramètres des Zones d'Interet,,,,,,,,,,,
 MoreStatsSettings,,,更多统计设置,,,More Stats Settings,,Plus de Paramètres de Statistiques,,,,,,,,,,,
 BetaSettings,,,测试设置,,,Beta Settings,,Paramètres en Beta,,,,,,,,,,,
+score,,,分数,,,score,,score,,,,,,,,,,,
+Time(seconds),,,时间(秒),,,Time(seconds),,Temps(secondes),,,,,,,,,,,
+Limiter Time,,,限制器时间,,,Limiter Time,,Limiteur de Temps,,,,,,,,,,,
```
Comment (Adrieennn) [inline gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv:245]: The translation for Chinese should be 时间限制器

File: gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv [hunk 5]
```diff
@@ -239,6 +240,10 @@ HeatMapSettings,,,眼动追踪地图设置,,,HeatMap Settings,,Paramètres de la
...
+Limiter Score,,,限制器得分,,,Limiter Score,,Limiteur de Score,,,,,,,,,,,
```
Comment (Adrieennn) [inline gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv:246]: 分数限制器 fits better here

File: gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv (hunk 4 above)
Comment (kuhlkrein) [reply gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv:245]: Thank you I'll change that :D

File: gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv (hunk 5 above)
Comment (kuhlkrein) [reply gazeplay-core/src/main/resources/data/multilinguism/multilinguism.csv:246]: Ok thanks

**Pull request merged**